monai/transforms/post/array.py

```diff
@@ -305,7 +305,11 @@ class KeepLargestConnectedComponent(Transform):
     backend = [TransformBackends.NUMPY]
 
     def __init__(
-        self, applied_labels: Union[Sequence[int], int], independent: bool = True, connectivity: Optional[int] = None
+        self,
+        applied_labels: Union[Sequence[int], int],
+        independent: bool = True,
+        connectivity: Optional[int] = None,
+        single_channel_onehot: bool = False,
```

Comment:
thanks, I think we can have an `is_one_hot` flag indicating whether the input will be interpreted as one-hot. `img.shape[0]` is never a reliable condition for this check...

Reply:
Hi @wyli ,

Thanks for your review and suggestion, adding this `is_one_hot` arg will make it breaking change, is it acceptable?

Reply:
I see, sorry I misunderstood this, I guess the actual feature request is the question of `applied_label` indexing from 0 or 1?

single-channel onehot is a bit confusing, it's only possible when num_classes is 1 and the data are all zeros, which is not very useful in practice...

Reply:
Hi @wyli ,

Or it can be the case as @holgerroth asked before, the data is in OneHot format and sliced with only 1 channel?
Anyway, I think you are right about "img.shape[0] is never a reliable condition for this check...", so I already totally updated in the latest commit to use `is_onehot` arg for it, I feel it's acceptable to change the API arg.
What do you think?

Thanks.